pkg/cmd/cli/cmd/debug.go

```diff
@@ -406,10 +406,13 @@ func (o *DebugOptions) Debug() error {
 		func() {
 			fmt.Fprintf(o.Attach.Err, "\nRemoving debug pod ...\n")
 			if err := o.Attach.Client.Pods(pod.Namespace).Delete(pod.Name, kapi.NewDeleteOptions(0)); err != nil {
-				fmt.Fprintf(o.Attach.Err, "error: unable to delete the debug pod %q: %v", pod.Name, err)
+				if !kapierrors.IsNotFound(err) {
+					fmt.Fprintf(o.Attach.Err, "error: unable to delete the debug pod %q: %v\n", pod.Name, err)
```

Comment:
You could successfully run a debug session, and when exiting the debug pod fails to be removed for any other reason - you should know the command left some garbage behind.